.pre-commit-config.yaml

```diff
@@ -0,0 +1,34 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pycqa/flake8
+    rev: '4.0.1'
+    hooks:
+    -   id: flake8
+        stages: [push]
+
+-   repo: https://github.com/pre-commit/mirrors-mypy
+    rev: 'v0.942'
+    hooks:
+    -   id: mypy
+        stages: [push]
+
+-   repo: https://github.com/pycqa/isort
+    rev: '5.10.1'
+    hooks:
+      - id: isort
+        name: isort (python)
+        stages: [push]
+
+# https://github.com/pre-commit/pre-commit/issues/761#issuecomment-394167542
+-   repo: local
+    hooks:
+    -   id: pytest
+        name: pytest
+        entry: venv/bin/pytest
+        language: script
+        pass_filenames: false
+        # alternatively you could `types: [python]` so it only runs when python files change
+        # though tests might be invalidated if you were to say change a data file
+        always_run: true
+        stages: [push]
```

Comment:
pytest is the one i'm not sure about ... 
pytest runs for ~2 minutes - and you can't really run it for "only changed files" - as it can't know which files affected which tests for certain (i doubt pre-commit tracks coverage by test and only runs what's necessary).

Therefore i think pytest should be either "optin" (so you have to do `git push --extra pytest` (if that's possible) - or not part of pre-commit at all.

If you know you're only fixing a documentation line (think - new contributor) - you'd not want to run 2 minutes of CI (you know you didn't change a code file).